### ISSUE TITLE
Space in parameters does not work for vertex ids

### ISSUE THREAD

**Issue by parkererickson** [pyTigerGraph/pyTigerGraph#22]
**Describe the bug**
Space in parameters passed to requests makes vertex id matching not work

**To Reproduce**
Search for a vertex id with a space in it.

**Expected behavior**
Returns vertex id with space in it.

https://stackoverflow.com/questions/21823965/use-20-instead-of-for-space-in-python-query-parameters/44829021

**Comment by szb-tg** [comment]
This is a [RFC1630](https://tools.ietf.org/html/rfc1630) (old standard) vs. [RFC3968](https://tools.ietf.org/html/rfc3986) (new standard) issue.
RFC1630 states "Within the query string, the plus sign is reserved as shorthand notation for a space", while RFC3968 (in [Section 2.1](https://tools.ietf.org/html/rfc3986#section-2.1)) says:

> A percent-encoding mechanism is used to represent a data octet in a component when that octet's corresponding character is outside the allowed set or is being used as a delimiter of, or within, the component.  A percent-encoded octet is encoded as a character triplet, consisting of the percent character "%" followed by the two hexadecimal digits representing that octet's numeric value.  **For example, "%20" is the percent-encoding for the binary octet "00100000" (ABNF: %x20), which in US-ASCII corresponds to the space character (SP)**.

The problem is that [`requests`](https://requests.readthedocs.io/en/master/) – used by pyTigerGraph – [calls `urllib.parse.urlencode()`](https://github.com/psf/requests/blob/master/requests/models.py#L105) (in 3.x) which by default uses [`quote_plus()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus) to encode spaces. As the call is deeply embedded in `requests`, we cannot pass an alternative function (such as [`quote()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote)) to `urlencode` from pyTigerGraph code to turn spaces to `%20`.

The issue was already reported (see link above), but the team behind `requests` is [unwilling to change behaviour](https://github.com/psf/requests/pull/5285#issuecomment-580970649).

Manually encoding space as `%20` in pyTigerGraph (before sending the URL params to `requests`) would generate `%2520` which in turn would mean literal `%20` to TigerGraph instead of space.
```
>>> print(urlencode({"alpha": "beta gamma"}))
alpha=beta+gamma
>>> print(urlencode({"alpha": "beta%20gamma"}))
alpha=beta%2520gamma
>>> print(urlencode({"alpha": "beta gamma"}, quote_via=quote))
alpha=beta%20gamma
>>> print(urlencode({"alpha": "beta%20gamma"}, quote_via=quote))
alpha=beta%2520gamma
```

We cannot expect TigerGraph (to be changed) to interpret plus sign as space since it complies with the current standard.

Not sure what can be done. Replacing `requests` with an alternative would be too complicated with potentially more problem introduced.

**Comment by parkererickson** [comment]
Fixed in v0.0.9.2
